recipes/poppler/all/conanfile.py

```diff
@@ -0,0 +1,265 @@
+            del self.options.enable_glib
+        if not self.options.get_safe("enable_glib"):
+            del self.options.enable_gobject_introspection
+            del self.options.enable_gtk
+        if self.options.fontconfiguration == "win32" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("'win32' option of fontconfig is only available on Windows")
+
+    def build_requirements(self):
+        self.build_requires("pkgconf/1.7.3")
+
+    def requirements(self):
+        self.requires("poppler-data/0.4.9")
+        self.requires("freetype/2.10.1")
+        if self.settings.os == "Windows":
+            self.requires("libiconv/1.16")
+        if self.options.fontconfiguration == "fontconfig":
+            self.require("fontconfig/2.13.91")
+        if self.options.get_safe("enable_glib"):
+            self.requires("glib/2.65.1")
+        if self.options.get_safe("enable_gobject_introspection"):
+            # FIXME: missing gobject_introspection recipe
+            raise ConanInvalidConfiguration("gobject_introspection is not (yet) available on cci")
+        if self.options.enable_qt:
+            # FIXME: missing qt recipe
+            raise ConanInvalidConfiguration("qt is not (yet) available on cii")
+        if self.options.get_safe("enable_gtk"):
+            # FIXME: missing gtk recipe
+            raise ConanInvalidConfiguration("gtk is not (yet) available on cii")
+        if self.options.enable_openjpeg:
+            self.requires("openjpeg/2.3.1")
+        if self.options.enable_lcms:
+            self.requires("lcms/2.11")
+        if self.options.enable_libjpeg:
+            self.requires("libjpeg/9d")
+        if self.options.enable_png:
+            self.requires("libpng/1.6.37")
+        if self.options.enable_nss:
+            # FIXME: missing nss recipe
+            raise ConanInvalidConfiguration("nss is not (yet) available on cii")
+        if self.options.enable_tiff:
+            self.requires("libtiff/4.1.0")
+        if self.options.enable_splash:
+            self.requires("boost/1.74.0")
+        if self.options.enable_libcurl:
+            self.requires("libcurl/7.71.1")
+        if self.options.enable_zlib:
+            self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("poppler-{}".format(self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["ENABLE_UNSTABLE_API_ABI_HEADERS"] = False
+        self._cmake.definitions["BUILD_GTK_TESTS"] = False
+        self._cmake.definitions["BUILD_QT5_TESTS"] = False
+        self._cmake.definitions["BUILD_QT6_TESTS"] = False
+        self._cmake.definitions["BUILD_CPP_TESTS"] = False
+
+        self._cmake.definitions["ENABLE_UTILS"] = False
+        self._cmake.definitions["ENABLE_CPP"] = True
```

Comment:
Is there any reason for not keep cpp as optional?

Reply:
Not that I think of.
Making it optional, and on by default would be ok too.
I just didn't bother here.